modules/kinesis/src/main/scala/com/snowplowanalytics/snowplow/enrich/kinesis/Sink.scala

```diff
@@ -224,6 +222,50 @@ object Sink {
       prre
     }
 
+  /**
+   * The result of trying to write a batch to kinesis
+   *  @param toRetry Records to retry, either because of throttling or an internal error
+   *  @param wasThrottled Whether at least one of retries is because of throttling
+   *  @param exampleInternalError A message to help with logging
+   */
+  private case class TryBatchResult(
+    toRetry: Vector[PutRecordsRequestEntry],
+    wasThrottled: Boolean,
+    exampleInternalError: Option[String]
+  ) {
+    def wasSuccessful: Boolean =
+      toRetry.isEmpty || wasThrottled
```

Comment:
We should **not** retry if:
1. At least **one** record has been successfully inserted (`records.size =!= prr.getFailedRecordCount.toInt`)
2. All the errors are `ProvisionedThroughputExceededException`

In all the other cases we should retry.

Is it the same than this? I'm not sure that we are catching the first condition.